torrt/bots/telegram_bot.py

```diff
@@ -0,0 +1,142 @@
+import logging
+
+from torrt.base_bot import BaseBot
+from torrt.toolbox import add_torrent_from_url
+from torrt.utils import TorrtConfig, BotClassesRegistry, get_torrent_from_url, RPCObjectsRegistry
+
+LOGGER = logging.getLogger(__name__)
+
+
+START, ASKING_URL, URL, PATH = 1, 2, 3, 4
+
+
+class TelegramBot(BaseBot):
+
+    alias = 'telegram'
+    url = 'https://api.telegram.org/bot'
+
+    def __init__(self, token, allowed_users=None):
+        """
+        :param token: str - Telegram's bot token
+        :param allowed_users: str - comma-joined list of users allowed to add new torrents.
+        """
+
+        self.token = token
+        allowed_users = allowed_users or ''
+        self.handler_kwargs = {}
+        try:
+            from telegram.ext import Updater, Filters
+            self.updater = Updater(token=self.token)
+            self.dispatcher = self.updater.dispatcher
+            if allowed_users:
+                self.handler_kwargs = {'filters': Filters.user(username=allowed_users.split(','))}
+
+        except ImportError:
+            LOGGER.error('You have not installed python-telegram-bot library.')
+            self.updater = None
+            self.dispatcher = None
+
+    def test_configuration(self):
+        return bool(self.updater)
+
+    def run(self):
+        self.add_handlers()
+        self.updater.start_polling()
+        self.updater.idle()
+
+    def add_handlers(self):
+        from telegram.ext import CommandHandler, ConversationHandler, RegexHandler, Filters, MessageHandler
+
+        conv_handler = ConversationHandler(
+            entry_points=[CommandHandler('start', start, **self.handler_kwargs)],
+
+            states={
+                ASKING_URL: [MessageHandler(Filters.text, asking_url_handler)],
+                URL: [RegexHandler('http[s]?://', url_handler, pass_user_data=True)],
+                PATH: [RegexHandler('/', path_handler, pass_user_data=True)],
+            },
+
+            fallbacks=[CommandHandler('cancel', cancel)]
+        )
+
+        self.dispatcher.add_handler(conv_handler)
+        self.dispatcher.add_handler(CommandHandler('add', add_torrent, **self.handler_kwargs))
+
+
+def asking_url_handler(bot, update):
```

Comment:
I'd rather expect those handler functions as methods to not to polute module namespace.